### PR TITLE
Subject 1. Make virtual data Infra and Get active user count in each month

### PR DESCRIPTION
**Pull request by shonjiho**
# Make virtual data Infra

### Background

My Company is developing incubator for cell. PM in our team have some problems.
When user open and close front door of incubator, current recovery speed of temperature, humidity in our incubator is so slow.

=> How to change humidifier power and heating?

For enhance this problem, We need state data of incubator as **raw_data**

**raw_data** : 
Temperatures for each internal area(Top, Center, Bottom), CO2, Humidity Suppling state, Heating state, experiment_id

**summary_table** : 
Results for each experiment.
- Time to reach target temperature
- Whether raise over humidity and temperature

# Get active user count in each month

```
 SELECT  DATE_TRUNC('month',ST.ts), COUNT(DISTINCT USC.userid) 
    FROM raw_data.user_session_channel as USC 
        INNER JOIN raw_data.session_timestamp as ST
        ON USC.sessionid = ST.sessionid 
    GROUP BY DATE_TRUNC('month',ST.ts)
    ORDER BY DATE_TRUNC('month',ST.ts);
```

- use  `INNER JOIN` for getting user sessions with timestamp 
- use `DATE_TRUNC` for getting month from datetime.
- use `COUNT(DISTINCT [column])` for removing duplicate userid in each month

### CODE REVIEW

File: subject-1/main.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    cursor = get_Redshift_connection()
+
+    # Query : count of active user in each month 
+    sql = """
+    SELECT  DATE_TRUNC('month',ST.ts), COUNT(DISTINCT USC.userid) 
+    FROM raw_data.user_session_channel as USC 
+        INNER JOIN raw_data.session_timestamp as ST
+        ON USC.sessionid = ST.sessionid 
+    GROUP BY DATE_TRUNC('month',ST.ts)
+    ORDER BY DATE_TRUNC('month',ST.ts);
```
Comment (keeyong) [inline subject-1/main.py:33]: Looks good. One way  to simplify GROUP BY and ORDER BY is to use the following:

```
GROUP BY 1
ORDER BY 1
```

File: subject-1/main.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    conn.set_session(readonly=True, autocommit=True)
+    return conn.cursor()
+
+
+if __name__ == "__main__":
+    cursor = get_Redshift_connection()
+
+    # Query : count of active user in each month 
+    sql = """
+    SELECT  DATE_TRUNC('month',ST.ts), COUNT(DISTINCT USC.userid)
```
Comment (keeyong) [inline subject-1/main.py:28]: Renaming the fields in SELECT can be convenient:

```
SELECT DATE_TRUNC('month',ST.ts) "month", COUNT(DISTINCT USC.userid) unique_users
```

File: subject-1/main.py (hunk 1 above)
Comment (jihoson94) [reply subject-1/main.py:33]: Thank for feedback. 

I knew it for the first time.

File: subject-1/main.py (hunk 2 above)
Comment (jihoson94) [reply subject-1/main.py:28]: Gotcha

**Pull request merged**